core/services/pipeline/task.mean.go

```diff
@@ -0,0 +1,70 @@
+package pipeline
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/shopspring/decimal"
+	"go.uber.org/multierr"
+)
+
+type MeanTask struct {
+	BaseTask      `mapstructure:",squash"`
+	Values        string `json:"values"`
+	AllowedFaults string `json:"allowedFaults"`
+	Precision     string `json:"precision"`
+}
+
+var _ Task = (*MeanTask)(nil)
+
+func (t *MeanTask) Type() TaskType {
+	return TaskTypeMean
+}
+
+func (t *MeanTask) Run(_ context.Context, vars Vars, _ JSONSerializable, inputs []Result) (result Result) {
+	var (
+		maybeAllowedFaults MaybeUint64Param
+		maybePrecision     MaybeUint64Param
+		valuesAndErrs      SliceParam
+		decimalValues      DecimalSliceParam
+		allowedFaults      int
+		faults             int
+	)
+	err := multierr.Combine(
+		errors.Wrap(ResolveParam(&maybeAllowedFaults, From(t.AllowedFaults)), "allowedFaults"),
+		errors.Wrap(ResolveParam(&maybePrecision, From(VarExpr(t.Precision, vars), t.Precision)), "precision"),
+		errors.Wrap(ResolveParam(&valuesAndErrs, From(VarExpr(t.Values, vars), JSONWithVarExprs(t.Values, vars, true), Inputs(inputs))), "values"),
+	)
+	if err != nil {
+		return Result{Error: err}
+	}
+
+	if allowed, isSet := maybeAllowedFaults.Uint64(); isSet {
+		allowedFaults = int(allowed)
+	} else {
+		allowedFaults = len(valuesAndErrs) - 1
+	}
+
+	values, faults := valuesAndErrs.FilterErrors()
+	if faults > allowedFaults {
+		return Result{Error: errors.Wrapf(ErrTooManyErrors, "Number of faulty inputs %v to mean task > number allowed faults %v", faults, allowedFaults)}
+	} else if len(values) == 0 {
+		return Result{Error: errors.Wrap(ErrWrongInputCardinality, "values")}
+	}
+
+	err = decimalValues.UnmarshalPipelineParam(values)
+	if err != nil {
+		return Result{Error: errors.Wrapf(ErrBadInput, "values: %v", err)}
+	}
+
+	total := decimal.NewFromInt(0)
+	for _, val := range decimalValues {
+		total = total.Add(val)
+	}
```

Comment:
Why not `decimal.Sum` ? https://pkg.go.dev/github.com/shopspring/decimal#Sum

Reply:
`decimal.Sum`'s signature is awkward if 1) you don't know at compile-time how many you'll be summing, and 2) you might only be passing it a single value. You would have to do this to be safe:
```go
total = decimalValues[0]
if len(decimalValues) > 1 {
    total = decimal.Sum(decimalValues[0], decimalValues[1:])
}
```
...which is the same number of lines.